gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaaas_driver.py

```diff
@@ -0,0 +1,418 @@
+
+    def _get_service_vendor(self, context, vpnservice_id):
+        vpnservice = self.service_plugin.get_vpnservice(
+                context, vpnservice_id)
+        desc = vpnservice['description']
+        # if the call is through GBP workflow,
+        # fetch the service profile from description
+        # else, use 'VYOS' as the service profile
+        if 'service_vendor=' in desc:
+            tokens = desc.split(';')
+            service_vendor = tokens[5].split('=')[1]
+        else:
+            service_vendor = 'VYOS'
+        return service_vendor
+
+    def create_ipsec_site_connection(self, context, ipsec_site_connection):
+
+        service_vendor = self._get_service_vendor(context,
+                          ipsec_site_connection['vpnservice_id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            ipsec_site_connection['vpnservice_id'],
+            rsrc_type='ipsec_site_connection',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=ipsec_site_connection['id'],
+            resource=ipsec_site_connection,
+            reason='create', service_vendor=service_vendor)
+
+    def update_ipsec_site_connection(self,
+                                     context,
+                                     old_ipsec_site_connection,
+                                     ipsec_site_connection):
+        service_vendor = self._get_service_vendor(context,
+                          ipsec_site_connection['vpnservice_id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            ipsec_site_connection['vpnservice_id'],
+            rsrc_type='ipsec_site_connection',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=ipsec_site_connection['id'],
+            resource={
+                'old': old_ipsec_site_connection,
+                'new': ipsec_site_connection},
+            reason='update', service_vendor=service_vendor)
+
+    def delete_ipsec_site_connection(self, context, ipsec_site_connection):
+        service_vendor = self._get_service_vendor(context,
+                          ipsec_site_connection['vpnservice_id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            ipsec_site_connection['vpnservice_id'],
+            rsrc_type='ipsec_site_connection',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=ipsec_site_connection['id'],
+            resource=ipsec_site_connection,
+            reason='delete', service_vendor=service_vendor)
+
+    def create_ikepolicy(self, context, ikepolicy):
+        pass
+
+    def delete_ikepolicy(self, context, ikepolicy):
+        pass
+
+    def update_ikepolicy(self, context, old_ikepolicy, ikepolicy):
+        pass
+
+    def create_ipsecpolicy(self, context, ipsecpolicy):
+        pass
+
+    def delete_ipsecpolicy(self, context, ipsecpolicy):
+        pass
+
+    def update_ipsecpolicy(self, context, old_ipsec_policy, ipsecpolicy):
+        pass
```

Comment:
Nothing to override in the following methods,
create_ikepolicy
delete_ikepolicy
update_ikepolicy
create_ipsecpolicy
delete_ipsecpolicy
update_ipsecpolicy

They can be removed.